### PR TITLE
Remove support for k8s 1.20.x

### PR DESCRIPTION
**Pull request by jesperpedersen**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -301,8 +300,8 @@ The `api` feature has been disabled, so the client API is not available. See ["C
 //!    With these macros, the two cases above would be solved like this:
 //!
 //!    - ```rust,ignore
-//!      // The compile_error!() is only emitted if 1.20 or lower is selected.
-//!      k8s_openapi::k8s_if_le_1_20! {
+//!      // The compile_error!() is only emitted if 1.21 or lower is selected.
+//!      k8s_openapi::k8s_if_le_1_21! {
 //!          compile_error!("This crate requires the v1_21 (or higher) feature to be enabled on the k8s-openapi crate.");
```
Comment (Arnavion) [inline src/lib.rs:305]: The `compile_error` message doesn't match the macro being used any more. And in fact the whole example doesn't make sense if 1.21 is the lowest supported version. Need to find something new that's supported in 1.22+ and not in 1.21

Comment (jesperpedersen) [reply src/lib.rs:305]: We should probably find something for 1.22 as 1.21 is EOL as well

Comment (Arnavion) [reply src/lib.rs:305]: Yes, the later the better, since that gives us more time before we have to find a new one.

Comment (jesperpedersen) [reply src/lib.rs:305]: I'll update the PR with 1.21 as well, and see if I can find something we can test upon

**Comment by Arnavion** [comment]
Just checked and ACK still supports v1.20 as of 2023-03-31, so this shouldn't be removed just yet.

Also I do these checks before I make a release anyway, so there's no reason for you and me to do extra work before that point.

**Comment by jesperpedersen** [comment]
Ok, np

**Pull request closed without merging**